### PR TITLE
M901

### PR DESCRIPTION
**Pull request by gwan93**
https://trello.com/c/io00FNjJ/901-change-how-the-checkboxes-work-in-the-filter-pane-and-add-hover-states

This PR:
- All methods are now default checked
- When a user checks/unchecks a method, the url query params will be updated with the remaining checked methods
- if no methods are checked, remove the method param from the url query params
- Make methods and projects clickable with a larger area
- Add background color and cursor pointer on hover to methods and projects
- Add some logic so that checking/unchecking a project will add/remove pins from the map

### CODE REVIEW

**Comment by gwan93** [comment]
[Change how the checkboxes work in the filter pane and add hover states](https://trello.com/c/io00FNjJ/901-change-how-the-checkboxes-work-in-the-filter-pane-and-add-hover-states)

File: src/components/FilterPane/FilterPane.styles.js [hunk 1]
```diff
@@ -120,3 +120,14 @@ export const StyledMenuItem = styled(MenuItem)`
     font-size: ${theme.typography.defaultFontSize};
   }
 `
+export const ExpandClickableArea = styled('div')`
```
Comment (JBurkinshaw) [inline src/components/FilterPane/FilterPane.styles.js:123]: This isn't following the `Styled...` naming convention. Is this intentional?
What do you think to something like `StyledCheckboxContainer` or similar?

File: src/components/LeafletMap.jsx [hunk 2]
```diff
@@ -114,8 +116,9 @@ const LeafletMap = ({ showFilterPane, showMetricsPane, view, setView }) => {
   const _addAndRemoveMarkersBasedOnFilters = useEffect(() => {
     const displayedProjectsChanged = displayedProjects !== prevDisplayedProjects
     const selectedMarkerChanged = selectedMarkerId !== prevSelectedMarkerId
+    const checkedProjectsChanged = checkedProjects !== prevCheckedProjects
```
Comment (JBurkinshaw) [inline src/components/LeafletMap.jsx:119]: I haven't tried this myself here, but I would assume that if `checkedProjects` is an array this condition would always be `true`. It checks whether the arrays are the same object in memory (which they're not), not whether their contents are identical. 

e.g.
```js
const array1 = [1, 2, 3]
const array2 = [1, 2, 3]

console.log(array1 === array2) // false
```

Consider writing a function or using a utility function to do a deep compare.

Comment (gwan93) [reply src/components/LeafletMap.jsx:119]: It seems to be doing a deep compare. I tried toggling one of the checkboxes and this is what I see in my console. If nothing changes then it is false, but when the checkboxes is toggled then it changes to true
image

**Review by JBurkinshaw — APPROVED** [review]

**Pull request merged**